taskflows/taskflows/__init__.py

```diff
@@ -259,20 +261,32 @@ def postprocess_description(task, _, input_, cluster, image, run_parameters, roo
         if item['name'] == 'description.json':
             files = list(client.listFile(item['_id']))
             if len(files) != 1:
-                raise Exception('Expecting a single file under item, found: %s' + len(files))
+                _log_std_err(task, client, description_folder)
+                msg = 'Expecting a single file under item, found: %s' % len(files)
+                _log_error(task, msg)
+                raise Exception(msg)
             description_file = files[0]
 
         elif item['name'] == 'pull.json':
             files = list(client.listFile(item['_id']))
             if len(files) != 1:
-                raise Exception('Expecting a single file under item, found: %s' + len(files))
+                _log_std_err(task, client, description_folder)
+                msg = 'Expecting a single file under item, found: %s' % len(files)
+                _log_error(task, msg)
+                raise Exception(msg)
             pull_file = files[0]
 
     if pull_file is None:
-        raise Exception('There was an error trying to pull the requested container image')
+        _log_std_err(task, client, description_folder)
+        msg = 'There was an error trying to pull the requested container image'
+        _log_error(task, msg)
+        raise Exception(msg)
 
     if description_file is None:
-        raise Exception('The container does not implement correctly the --description flag')
+        _log_std_err(task, client, description_folder)
+        msg = 'The container does not implement correctly the --description flag'
+        _log_error(task, msg)
+        raise Exception(msg)
```

Comment:
Seems like a small function would remove some of the boiler plate of logging the errors and raising the exception?